flask/app/datasets.py

```diff
@@ -163,31 +170,15 @@ def list_datasets(page: int, limit: int) -> Response:
             .filter(models.users_groups_table.columns.user_id == user_id, *filters)
         )
     else:  # Admin or LOGIN_DISABLED, authorized to query all datasets
-        query = (
-            models.Dataset.query.options(
-                contains_eager(models.Dataset.tissue_sample)
-                .contains_eager(models.TissueSample.participant)
-                .contains_eager(models.Participant.family),
-                contains_eager(models.Dataset.tissue_sample)
-                .contains_eager(models.TissueSample.participant)
-                .joinedload(models.Participant.institution),
-                contains_eager(models.Dataset.files),
-                contains_eager(models.Dataset.updated_by),
-            )
-            .join(models.Dataset.tissue_sample)
-            .outerjoin(models.Dataset.groups)
-            .join(models.TissueSample.participant)
-            .join(models.Participant.family)
-            .outerjoin(models.Dataset.files)
-            .join(models.Dataset.updated_by)
-            .filter(*filters)
-        )
+        query = query.options(joinedload(models.Dataset.groups)).filter(*filters)
 
     total_count = query.with_entities(
         func.count(distinct(models.Dataset.dataset_id))
     ).scalar()
 
-    datasets = query.order_by(order).limit(limit).offset(page * (limit or 0)).all()
+    datasets = (
+        query.distinct().order_by(*order).limit(limit).offset(page * (limit or 0)).all()
```

Comment:
i'm still a bit confused about the distinct call as (unless i'm misunderstanding) it seems to me the resulting rows from the query would already be unique? i tried omitting the distinct call from the query and all the tests still pass.

Reply:
I think I figured it out. The distinct call was put in-place first when we were doing an outer join by groups. This reduced the number of rows in the query such that only 1 of each dataset_id occurred.

Since then, we've replaced the outer join by groups with a joinedload by groups, eliminating the need for the distinct call at all since groups aren't part of the query. So we accidentally found 2 ways to solve the problem: outer-joining groups and using distinct, or eager-loading groups.